scripts/rename_classes.py

```
@@ -0,0 +1,252 @@
+    Build regex patterns for each class name to match different contexts where they might appear.
+    Avoids matching in file paths by excluding patterns followed by a period.
+    """
+    patterns = {}
+    for old_name in CLASS_MAPPING:
+        # This pattern will match class definitions and usages but not when followed by a period,
+        # which would indicate a file extension or namespace access
+        patterns[old_name] = re.compile(r'\b' + re.escape(old_name) + r'\b(?!\.)')
+    return patterns
+
+
+def should_skip_file(file_path: str, extensions: List[str]) -> bool:
```

Ruff (0.8.2)

Use `list` instead of `List` for type annotation

Replace with `list`

(UP006)